Description: null-terminate long names of processes, otherwise a crash
 can occur
Author: Mikko Rapeli <mikko.rapeli@iki.fi>
Bug-Debian: http://bugs.debian.org/682388
Forwarded: not-needed
Last-Update: 20130421

--- powertop-2.0.orig/src/process/process.cpp
+++ powertop-2.0/src/process/process.cpp
@@ -86,7 +86,7 @@
 
 process::process(const char *_comm, int _pid, int _tid) : power_consumer()
 {
-	char line[4096];
+	char line[4097];
 	ifstream file;
 
 	strcpy(comm, _comm);
@@ -103,6 +103,7 @@
 		file.open(line);
 		while (file) {
 			file.getline(line, 4096);
+			line[4096] = '\0';
 			if (strstr(line, "Tgid")) {
 				char *c;
 				c = strchr(line, ':');
